src/PhpSpec/ServiceContainer.php

```diff
@@ -21,26 +26,32 @@
  */
 class ServiceContainer
 {
+    const COMPAT_SERVICE_FACTORY_ID = 'compat.service-factory';
+    private $serviceFactory;
     /**
-     * @var array
-     */
-    private $parameters = array();
-
-    /**
-     * @var array
-     */
-    private $services = array();
-
-    /**
-     * @var array
+     * @var ContainerBuilder
      */
-    private $prefixed = array();
+    private $containerBuilder;
+    private $params = array();
+    private $defaultExtensions = array();
 
     /**
      * @var array
      */
     private $configurators = array();
 
+    public function __construct($container)
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('paths.base', __DIR__);
+
+        $this->containerBuilder = $container;
+        $definition = new Definition('PhpSpec\\ServiceFactory');
+        $this->containerBuilder->setDefinition(self::COMPAT_SERVICE_FACTORY_ID, $definition);
+        $this->serviceFactory = $this->containerBuilder->get(self::COMPAT_SERVICE_FACTORY_ID);
```

Comment:
For me the question is whether it causes any issues for existing extensions, and whether it would encourage any bad practice going forwards in future when people write extensions only using the new interface.